### PR TITLE
Support multi-device configuration for MKL-DNN

### PR DESCRIPTION
**Pull request by PenghuiCheng**
MKL-DNN support multi-node mode，but not support multi-devices mode,this commit will support multi-devices for MKL-DNN.This commit  depend on https://github.com/pytorch/pytorch/pull/11330

### CODE REVIEW

**Comment by jspisak** [comment]
cc @yinghai

**Comment by PenghuiCheng** [comment]
@yinghai，I had rebased code.Please review it.
thanks!

**Review by yinghai — APPROVED** [review]
LGTM

File: caffe2/python/data_parallel_model.py [hunk 1]
```diff
@@ -823,8 +823,12 @@ def builder_fun(model):
 
     if device is None:
         device = scope.CurrentDeviceScope()
-
-    device_prefix = "gpu" if core.IsGPUDeviceType(device.device_type) else "cpu"
+    if device.device_type == caffe2_pb2.CUDA:
```
Comment (bddppq) [inline caffe2/python/data_parallel_model.py]: Here should use core.IsGPUDeviceType instead of directly comparing to CUDA

**Review by bddppq — CHANGES_REQUESTED** [review]

**Comment by PenghuiCheng** [comment]
@bddppq , @yinghai ,I had updated codes as your suggestion,Please review it.
Thanks!

File: caffe2/python/data_parallel_model.py (hunk 1 above)
Comment (PenghuiCheng) [reply caffe2/python/data_parallel_model.py]: Had updated.

**Comment by yinghai** [comment]
@bddppq Could you take a look?

**Pull request closed without merging**